### ISSUE TITLE
bump pyyaml to 6.0.1

### ISSUE THREAD

**Issue by chenrui333** [anibali/pywebp#41]
relates to https://github.com/yaml/pyyaml/issues/601
also relates to https://github.com/Homebrew/homebrew-core/pull/135038

**Comment by chenrui333** [comment]
- relates to https://github.com/conan-io/conan/pull/13990

**Comment by anibali** [comment]
Please give a proper description of the issue.

**Comment by chenrui333** [comment]
@anibali it is basically the build problem for pywebp to build with pyyaml against cython 3+, which is fixed in pyyaml 6.0.1, let me know if that makes sense. Thanks!

**Comment by chenrui333** [comment]
This is the error build log if that helps
<details>
<summary>error build log</summary>

```
Processing /private/tmp/ddev--webp-20230719-93379-o7v50c/webp-0.1.6
  Installing build dependencies: started
  Running command pip subprocess to install build dependencies
  Collecting cffi>=1.0.0
    Using cached cffi-1.15.1-cp311-cp311-macosx_11_0_arm64.whl
  Collecting conan<2.0,>=1.8.0
    Downloading conan-1.60.1.tar.gz (781 kB)
       ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 781.7/781.7 kB 36.2 MB/s eta 0:00:00
    Preparing metadata (setup.py): started
    Preparing metadata (setup.py): finished with status 'done'
  Collecting poetry-core>=1.2.0
    Using cached poetry_core-1.6.1-py3-none-any.whl
  Collecting setuptools>=40.8.0
    Using cached setuptools-68.0.0-py3-none-any.whl
  Collecting wheel
    Using cached wheel-0.40.0-py3-none-any.whl
  Collecting pycparser (from cffi>=1.0.0)
    Using cached pycparser-2.21-py2.py3-none-any.whl
  Collecting requests<3.0.0,>=2.25 (from conan<2.0,>=1.8.0)
    Using cached requests-2.31.0-py3-none-any.whl
  Collecting urllib3<1.27,>=1.26.6 (from conan<2.0,>=1.8.0)
    Using cached urllib3-1.26.16-py2.py3-none-any.whl
  Collecting colorama<0.5.0,>=0.3.3 (from conan<2.0,>=1.8.0)
    Downloading colorama-0.4.6.tar.gz (27 kB)
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'done'
    Preparing metadata (pyproject.toml): started
    Preparing metadata (pyproject.toml): finished with status 'done'
  Collecting PyYAML<=6.0,>=3.11 (from conan<2.0,>=1.8.0)
    Downloading PyYAML-6.0.tar.gz (124 kB)
       ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 125.0/125.0 kB 15.3 MB/s eta 0:00:00
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'error'
    error: subprocess-exited-with-error
 
    × Getting requirements to build wheel did not run successfully.
    │ exit code: 1
    ╰─> [54 lines of output]
        running egg_info
        writing lib/PyYAML.egg-info/PKG-INFO
        writing dependency_links to lib/PyYAML.egg-info/dependency_links.txt
        writing top-level names to lib/PyYAML.egg-info/top_level.txt
        Traceback (most recent call last):
          File "/opt/homebrew/Cellar/ddev/3.1.0/libexec/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
            main()
          File "/opt/homebrew/Cellar/ddev/3.1.0/libexec/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
            json_out['return_val'] = hook(**hook_input['kwargs'])
                                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          File "/opt/homebrew/Cellar/ddev/3.1.0/libexec/lib/python3.11/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
            return hook(config_settings)
                   ^^^^^^^^^^^^^^^^^^^^^
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 341, in get_requires_for_build_wheel
            return self._get_build_requires(config_settings, requirements=['wheel'])
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 323, in _get_build_requires
            self.run_setup()
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/build_meta.py", line 338, in run_setup
            exec(code, locals())
          File "<string>", line 288, in <module>
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/__init__.py", line 107, in setup
            return distutils.core.setup(**attrs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 185, in setup
            return run_commands(dist)
                   ^^^^^^^^^^^^^^^^^^
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/_distutils/core.py", line 201, in run_commands
            dist.run_commands()
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 969, in run_commands
            self.run_command(cmd)
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/dist.py", line 1234, in run_command
            super().run_command(command)
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/_distutils/dist.py", line 988, in run_command
            cmd_obj.run()
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 314, in run
            self.find_sources()
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 322, in find_sources
            mm.run()
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 551, in run
            self.add_defaults()
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/command/egg_info.py", line 589, in add_defaults
            sdist.add_defaults(self)
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/command/sdist.py", line 104, in add_defaults
            super().add_defaults()
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/sdist.py", line 251, in add_defaults
            self._add_defaults_ext()
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/_distutils/command/sdist.py", line 336, in _add_defaults_ext
            self.filelist.extend(build_ext.get_source_files())
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
          File "<string>", line 204, in get_source_files
          File "/private/tmp/pip-build-env-9xtz90bu/overlay/lib/python3.11/site-packages/setuptools/_distutils/cmd.py", line 107, in __getattr__
            raise AttributeError(attr)
        AttributeError: cython_sources
        [end of output]
 
    note: This error originates from a subprocess, and is likely not a problem with pip.
  error: subprocess-exited-with-error
 
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
 
  note: This error originates from a subprocess, and is likely not a problem with pip.
```

</details>

here is the full build log, https://github.com/Homebrew/homebrew-core/actions/runs/5588641673/job/15150254346

**Comment by anibali** [comment]
The upstream issue is here: https://github.com/yaml/pyyaml/issues/724

The immediate, short-term fix would be to force using pyyaml 5.3.1, e.g. by blacklisting broken versions: https://github.com/yaml/pyyaml/issues/724#issuecomment-1638591821. However, my preference is to wait for an upstream fix and then update the poetry lock file appropriately without changing immediate dependencies of pywebp.

**Comment by anibali** [comment]
Alternatively, I am now also watching https://github.com/conan-io/conan/issues/14319 which would provide a path forward.
